core/consensus/production_consensus.hpp

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include "consensus/timeline/types.hpp"
+#include "primitives/block_header.hpp"
+
+namespace kagome::consensus {
+
+  enum class ValidatorStatus {
+    NonValidator = 0,
+    InactiveValidator,
+    Validator,
+    SingleValidator,
+  };
+
+  class ProductionConsensus {
```

Comment:
Drop few comments explaining that this is block production consensus. Otherwise term "production" be confusing here.
Maybe it even makes sense to rename to `BlockProductionConsensus`